stubs/paho-mqtt/paho/mqtt/client.pyi

```diff
@@ -0,0 +1,321 @@
+_ExtraHeader: TypeAlias = dict[str, str] | Callable[[dict[str, str]], dict[str, str]]
+_OnLog: TypeAlias = Callable[[Client, Any, int, str], None]
+_OnConnect = TypeVar(
+    "_OnConnect",
+    Callable[[Client, Any, dict[str, int], int], None],
+    Callable[[Client, Any, dict[str, int], ReasonCodes, Properties | None], None],
+)
+_OnConnectFail: TypeAlias = Callable[[Client, Any], None]
+_OnSubscribe = TypeVar(
+    "_OnSubscribe",
+    Callable[[Client, Any, int, Iterable[int]], None],
+    Callable[[Client, Any, Iterable[ReasonCodes], Iterable[Properties]], None],
+)
+_OnMessage: TypeAlias = Callable[[Client, Any, MQTTMessage], None]
+_OnPublish: TypeAlias = Callable[[Client, Any, int], None]
+_OnUnsubscribe = TypeVar(
+    "_OnUnsubscribe",
+    Callable[[Client, Any, int], None],
+    Callable[[Client, Any, int, Iterable[Properties], Iterable[ReasonCodes]], None],
+)
+_OnDisconnect = TypeVar(
+    "_OnDisconnect", Callable[[Client, Any, int], None], Callable[[Client, Any, int, ReasonCodes | None, Properties | None], None]
+)
+_OnSocket: TypeAlias = Callable[[Client, Any, _socket.socket | Any | None], None]
+
+class WebsocketConnectionError(ValueError): ...
+
+def error_string(mqtt_errno: int) -> str: ...
+def connack_string(connack_code: int) -> str: ...
+def base62(num: int, base: str = ..., padding: int = ...) -> str: ...
+def topic_matches_sub(sub: str, topic: str) -> bool: ...
+
+class MQTTMessageInfo:
+    mid: int
+    rc: int
+    def __init__(self, mid: int) -> None: ...
+    def __iter__(self) -> MQTTMessageInfo: ...
+    def __next__(self) -> int: ...
+    def next(self) -> int: ...
+    def __getitem__(self, index: int) -> int: ...
+    def wait_for_publish(self, timeout: float | None = ...) -> None: ...
+    def is_published(self) -> bool: ...
+
+class MQTTMessage:
+    timestamp: int
+    state: int
+    dup: bool
+    mid: int
+    payload: bytes | bytearray
+    qos: int
+    retain: bool
+    info: MQTTMessageInfo
+    properties: Properties | None
+    def __init__(self, mid: int = ..., topic: bytes | bytearray = ...) -> None: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+    @property
+    def topic(self) -> str: ...
+    @topic.setter
+    def topic(self, value: bytes | bytearray) -> None: ...
+
+class Client:
+    suppress_exceptions: bool
+    def __init__(
+        self,
+        client_id: str = ...,
+        clean_session: bool | None = ...,
+        userdata: Any | None = ...,
+        protocol: int = ...,
+        transport: str = ...,
+        reconnect_on_failure: bool = ...,
+    ) -> None: ...
+    def __del__(self) -> None: ...
+    def reinitialise(self, client_id: str = ..., clean_session: bool = ..., userdata: Any | None = ...) -> None: ...
+    def ws_set_options(self, path: str = ..., headers: _ExtraHeader | None = ...) -> None: ...
+    def tls_set_context(self, context: Any | None = ...) -> None: ...
```

Comment:
```suggestion
    def tls_set_context(self, context: _ssl.Context | None = ...) -> None: ...
```

Needs `import ssl as _ssl` at the top of the file.

Reply:
Is it `_ssl.SSLContext` and not `_ssl.Context`?

Reply:
Yes, `ssl.Context` does not exist:

```
>>> import ssl
>>> ssl.Context
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'ssl' has no attribute 'Context'
>>> ssl.SSLContext
<class 'ssl.SSLContext'>
```